### PR TITLE
Enable mypy in CI and add full type annotation for config and ima AST

### PR DESCRIPTION
**Pull request by THS-on**
- tox: add mypy support
- json: allow ignore comment to be parsed by mypy
- ima/ast: add missing type hints
- config: add missing type hints
- ci: add mypy to CI

### CODE REVIEW

**Comment by stefanberger** [comment]
LGTM

**Review by aplanas — APPROVED** [review]

**Comment by stefanberger** [comment]
> I needed to relax the type for the `yaml_to_dict(.. )` function, so that pyright does not complain.

After I saw your changes I was wondering how you got past pyright...

**Comment by THS-on** [comment]
> After I saw your changes I was wondering how you got past pyright...

I changed the type hint for `arry` to `Any`. This is not ideal, but I couldn't figure out another way to to make pyright happy.

**Review by aplanas — APPROVED** [review]

File: test-requirements.txt [hunk 1]
```diff
@@ -1,2 +1,10 @@
 dbus-python
-jinja2
```
Comment (mpeters) [inline test-requirements.txt:2]: Was this used during the config conversion? And we're no longer doing that during tests? Do we need a test for that specifically?

**Review by mpeters — APPROVED** [review]
One small question, but doesn't hold up the approval

**Pull request merged**